src/bmffimage.cpp

```diff
@@ -184,7 +184,7 @@ long BmffImage::boxHandler(std::ostream& out /* = std::cout*/, Exiv2::PrintStruc
   if (bTrace) {
     bLF = true;
     out << indent(depth) << "Exiv2::BmffImage::boxHandler: " << toAscii(box_type)
-        << Internal::stringFormat(" %8ld->%lu ", address, box_length);
+        << Internal::stringFormat(" %8ld->%llu ", address, box_length);
```

Comment:
Even better, let's use the [<cinttypes> macros](https://www.cplusplus.com/reference/cinttypes/) and forget portability issues popping back here.

Reply:
I'll give it a try for the first time in my life